schema-registry/authorizer/authorization-agent/src/main/java/com/hortonworks/registries/schemaregistry/authorizer/agent/AuthorizationUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016-2020 Cloudera, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry.authorizer.agent;
+
+import com.hortonworks.registries.auth.util.KerberosName;
+import com.hortonworks.registries.schemaregistry.authorizer.core.Authorizer;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import javax.ws.rs.core.SecurityContext;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AuthorizationUtils {
+
+    private static Map<String, Authorizer.UserAndGroups> userGroupsStore = new ConcurrentHashMap<>();
+
+    public static Authorizer.UserAndGroups getUserAndGroups(SecurityContext sc) {
+
+        Principal p = sc.getUserPrincipal();
+        if(p == null) {
+            return null;
+        }
+        KerberosName kerberosName = new KerberosName(p.getName());
```

Comment:
@vomoshkovskyi @raju-saravanan
I think this should not happen. Kerberos specific parsing should have been already taken care of, at ServletFilter level or at SecurityContext creation time. Else you would end up performing authentication specific user extraction here.

Reply:
@guruchai How this should be modified? Could you, please, provide an example.